engine/src/snapshot/migration/mod.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod error;
+mod v2;
+mod v3;
+
+use std::{
+    fs::{File, OpenOptions},
+    io::{Read, Write},
+    path::Path,
+};
+
+use crypto::{
+    ciphers::{chacha::XChaCha20Poly1305, traits::Aead},
+    hashes::{blake2b, Digest},
+    keys::{age, x25519},
+};
+
+// These dependencies must not change between versions,
+// otherwise migration will work differently.
+pub use self::error::Error;
+use crate::snapshot::{compress, decompress};
+use zeroize::Zeroizing;
+
+pub enum Version<'a> {
+    V2 {
+        path: &'a Path,
+        key: &'a [u8; 32],
+        aad: &'a [u8],
+    },
+    V3 {
+        path: &'a Path,
+        password: &'a [u8],
+    },
+}
+
+impl<'a> Version<'a> {
+    pub fn v2(path: &'a Path, key: &'a [u8; 32], aad: &'a [u8]) -> Self {
+        Self::V2 { path, key, aad }
+    }
+
+    pub fn v3(path: &'a Path, password: &'a [u8]) -> Self {
+        Self::V3 { path, password }
+    }
+}
+
+/// Magic bytes (bytes 0-4 in a snapshot file) aka PARTI
+const MAGIC: [u8; 5] = [0x50, 0x41, 0x52, 0x54, 0x49];
```

Comment:
Any specific reason why it's exactly these and not something else?

Reply:
Nvm is there for historic reasons

Reply:
I have no clue. That's the old magic bytes.